<a href="https://colab.research.google.com/github/TongLuo2021/HPO-DistilBert-With-Ray-and-Optuna/blob/main/HPO_DistilBert_with_Ray_and_Optuna.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Hyperparameter Search with Transformers and Ray Tune, and Optuna
* by Tong Luo

Hyperparameters (HPs) can have significant impact on the final model parameters and model performance. [Study show](https://towardsdatascience.com/hyperparameter-optimization-for-optimum-transformer-models-b95a32b70949) that the most significant HPs are learning rate and training epochs, while other parameters also matters. Depends on the applications such as CV, NLP, the HPs can be significantly.

----------
###TLuo's works are:
1. Ray and Optuna has [default HPs]((https://github.com/huggingface/transformers/blob/main/src/transformers/trainer_utils.py): 
* learning_rate, 
* num_train_epochs, 
* seed, 
* per_device_train_batch_size. 
2. Besides that, I used customized HPO, by defining hp_space_ray function following the [source code](https://github.com/huggingface/transformers/blob/main/src/transformers/trainer_utils.py), and [sgugers's example](https://github.com/huggingface/transformers/blob/main/src/transformers/trainer_utils.py)
3. Uses same frame work for Optuna.
4. Use same frame work, we can tune Other hyper parameters are listed [here](https://simpletransformers.ai/docs/usage/#configuring-a-simple-transformers-model) .
5. For small HPO setting (n_runtime = 3), the total time is similiar. Ray suppose to be faster because it can parallel moltiple tasks. More experiment need to be done here. 
----------
###[References]
* [Hyperparameter Search with Transformers and Ray Tune](https://huggingface.co/blog/ray-tune)
* [Hyperparameter optimization for optimun transformer models](https://towardsdatascience.com/hyperparameter-optimization-for-optimum-transformer-models-b95a32b70949)


## Choose which HPO you want to run, default to RUN_BOTH

In [1]:
RUN_RAY = 0
RUN_OPTUNA = 1
RUN_ALL = 2

# Make your selection here
RUN = RUN_ALL

In [2]:
pip install "ray[tune]" transformers datasets scipy sklearn torch

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 59.1 MB 1.4 MB/s 
     |████████████████████████████████| 5.5 MB 67.3 MB/s 
     |████████████████████████████████| 451 kB 69.1 MB/s 
     |████████████████████████████████| 182 kB 91.7 MB/s 
     |████████████████████████████████| 7.6 MB 78.6 MB/s 
     |████████████████████████████████| 212 kB 94.8 MB/s 
     |████████████████████████████████| 115 kB 91.6 MB/s 
     |████████████████████████████████| 127 kB 94.5 MB/s 
     |████████████████████████████████| 8.8 MB 67.4 MB/s 
     |████████████████████████████████| 125 kB 96.5 MB/s 
     |████████████████████████████████| 468 kB 77.8 MB/s 
  Created wheel for sklearn: filename=sklearn-0.0.post1-py3-none-any.whl size=2344 sha256=05ada864833551db8d5bc9fb12794f66d01cfc3e5b4e3ae2780a9c716e7cc329
  Stored in directory: /root/.cache/pip/wheels/42/56/cc/4a8bf86613aafd5b7f1b310477667c1fca5c51c3ae4124a003
S

In [3]:
from datasets import load_dataset, load_metric
from transformers import (AutoModelForSequenceClassification, AutoTokenizer,
                          Trainer, TrainingArguments)

In [4]:
tokenizer = AutoTokenizer.from_pretrained('distilbert-base-uncased')
dataset = load_dataset('glue', 'mrpc')
metric = load_metric('glue', 'mrpc')

def encode(examples):
    outputs = tokenizer(
        examples['sentence1'], examples['sentence2'], truncation=True)
    return outputs

encoded_dataset = dataset.map(encode, batched=True)

def model_init():
    return AutoModelForSequenceClassification.from_pretrained(
        'distilbert-base-uncased', return_dict=True)

def compute_metrics(eval_pred):
    predictions, labels = eval_pred
    predictions = predictions.argmax(axis=-1)
    return metric.compute(predictions=predictions, references=labels)

# Evaluate during training and a bit more often
# than the default to be able to prune bad trials early.
# Disabling tqdm is a matter of preference.
training_args = TrainingArguments(
    "test", 
    evaluation_strategy="steps", 
    eval_steps=500, 
    disable_tqdm=False)

trainer = Trainer(
    args=training_args,
    tokenizer=tokenizer,
    train_dataset=encoded_dataset["train"],
    eval_dataset=encoded_dataset["validation"],
    model_init=model_init,
    compute_metrics=compute_metrics,
)





Downloading:   0%|          | 0.00/28.0 [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/483 [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/232k [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/466k [00:00<?, ?B/s]

Generating train split:   0%|          | 0/3668 [00:00<?, ? examples/s]

Generating validation split:   0%|          | 0/408 [00:00<?, ? examples/s]

Generating test split:   0%|          | 0/1725 [00:00<?, ? examples/s]

Dataset glue downloaded and prepared to /root/.cache/huggingface/datasets/glue/mrpc/1.0.0/dacbe3125aa31d7f70367a07a8a9e72a5a0bfeb5fc42e75c9db75b96da6053ad. Subsequent calls will reuse this data.


  0%|          | 0/3 [00:00<?, ?it/s]

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:3: FutureWarning: load_metric is deprecated and will be removed in the next major version of datasets. Use 'evaluate.load' instead, from the new library 🤗 Evaluate: https://huggingface.co/docs/evaluate
  This is separate from the ipykernel package so we can avoid doing imports until


  0%|          | 0/4 [00:00<?, ?ba/s]

  0%|          | 0/1 [00:00<?, ?ba/s]

  0%|          | 0/2 [00:00<?, ?ba/s]

loading configuration file config.json from cache at /root/.cache/huggingface/hub/models--distilbert-base-uncased/snapshots/1c4513b2eedbda136f57676a34eea67aba266e5c/config.json
Model config DistilBertConfig {
  "_name_or_path": "distilbert-base-uncased",
  "activation": "gelu",
  "architectures": [
    "DistilBertForMaskedLM"
  ],
  "attention_dropout": 0.1,
  "dim": 768,
  "dropout": 0.1,
  "hidden_dim": 3072,
  "initializer_range": 0.02,
  "max_position_embeddings": 512,
  "model_type": "distilbert",
  "n_heads": 12,
  "n_layers": 6,
  "pad_token_id": 0,
  "qa_dropout": 0.1,
  "seq_classif_dropout": 0.2,
  "sinusoidal_pos_embds": false,
  "tie_weights_": true,
  "transformers_version": "4.24.0",
  "vocab_size": 30522
}



Downloading:   0%|          | 0.00/268M [00:00<?, ?B/s]

loading weights file pytorch_model.bin from cache at /root/.cache/huggingface/hub/models--distilbert-base-uncased/snapshots/1c4513b2eedbda136f57676a34eea67aba266e5c/pytorch_model.bin
Some weights of the model checkpoint at distilbert-base-uncased were not used when initializing DistilBertForSequenceClassification: ['vocab_layer_norm.bias', 'vocab_transform.weight', 'vocab_projector.weight', 'vocab_projector.bias', 'vocab_transform.bias', 'vocab_layer_norm.weight']
- This IS expected if you are initializing DistilBertForSequenceClassification from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing DistilBertForSequenceClassification from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
Some weights of DistilBert

##Using Ray to customize the hyper parameters

In [5]:
from typing import Dict

def hp_space_ray(trial) -> Dict[str, float]:
  from ray import tune
  return {
      "learning_rate": tune.loguniform(1e-6, 1e-4),
      "num_train_epochs": tune.choice(list(range(1, 6))),
      "seed": tune.uniform(1, 40),
      "per_device_train_batch_size": tune.choice([4, 8, 16, 32, 64]),
      "weight_decay": tune.choice([0.1, 0.3, 0.03])
  }


* un-comment this block to run Ray HPO

In [6]:
# Default objective is the sum of all metrics
# when metrics are provided, so we have to maximize it.
%%time
if RUN == RUN_RAY or RUN == RUN_ALL:
  best_run = trainer.hyperparameter_search(
    hp_space = hp_space_ray,
    direction="maximize", 
    backend="ray", 
    n_trials=3)
  print("------------RAY------------")
  print(best_run)

No `resources_per_trial` arg was passed into `hyperparameter_search`. Setting it to a default value of 1 CPU and 1 GPU for each trial.
2022-11-29 17:25:34,506	INFO worker.py:1528 -- Started a local Ray instance.
/usr/local/lib/python3.7/dist-packages/ray/tune/trainable/function_trainable.py:611: DeprecationWarning: `checkpoint_dir` in `func(config, checkpoint_dir)` is being deprecated. To save and load checkpoint in trainable functions, please use the `ray.air.session` API:

from ray.air import session

def train(config):
    # ...
    session.report({"metric": metric}, checkpoint=checkpoint)

For more information please see https://docs.ray.io/en/master/tune/api_docs/trainable.html

  DeprecationWarning,


== Status ==
Current time: 2022-11-29 17:25:38 (running for 00:00:00.19)
Memory usage on this node: 3.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (2 PENDING, 1 RUNNING)
+------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------+
| Trial name             | status   | loc            |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |
|                        |          |                |                 |                    |                ch_size |         |                |
|------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------|
| _objective_d6e35_00000 | RUNNING  | 172.2

(_objective pid=659) Some weights of the model checkpoint at distilbert-base-uncased were not used when initializing DistilBertForSequenceClassification: ['vocab_layer_norm.bias', 'vocab_layer_norm.weight', 'vocab_transform.weight', 'vocab_projector.weight', 'vocab_transform.bias', 'vocab_projector.bias']
(_objective pid=659) - This IS expected if you are initializing DistilBertForSequenceClassification from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
(_objective pid=659) - This IS NOT expected if you are initializing DistilBertForSequenceClassification from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
(_objective pid=659) Some weights of DistilBertForSequenceClassification were not initialized from the model checkpoint at distilbert-base-uncased

== Status ==
Current time: 2022-11-29 17:25:47 (running for 00:00:09.92)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (2 PENDING, 1 RUNNING)
+------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------+
| Trial name             | status   | loc            |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |
|                        |          |                |                 |                    |                ch_size |         |                |
|------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------|
| _objective_d6e35_00000 | RUNNING  | 172.2

  2%|▏         | 7/290 [00:05<02:13,  2.12it/s]


== Status ==
Current time: 2022-11-29 17:25:53 (running for 00:00:14.92)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (2 PENDING, 1 RUNNING)
+------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------+
| Trial name             | status   | loc            |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |
|                        |          |                |                 |                    |                ch_size |         |                |
|------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------|
| _objective_d6e35_00000 | RUNNING  | 172.2

  7%|▋         | 20/290 [00:10<01:45,  2.56it/s]


== Status ==
Current time: 2022-11-29 17:25:58 (running for 00:00:19.93)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (2 PENDING, 1 RUNNING)
+------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------+
| Trial name             | status   | loc            |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |
|                        |          |                |                 |                    |                ch_size |         |                |
|------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------|
| _objective_d6e35_00000 | RUNNING  | 172.2

 11%|█▏        | 33/290 [00:15<01:43,  2.48it/s]


== Status ==
Current time: 2022-11-29 17:26:03 (running for 00:00:24.93)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (2 PENDING, 1 RUNNING)
+------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------+
| Trial name             | status   | loc            |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |
|                        |          |                |                 |                    |                ch_size |         |                |
|------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------|
| _objective_d6e35_00000 | RUNNING  | 172.2

 16%|█▌        | 46/290 [00:20<01:32,  2.64it/s]


== Status ==
Current time: 2022-11-29 17:26:08 (running for 00:00:29.94)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (2 PENDING, 1 RUNNING)
+------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------+
| Trial name             | status   | loc            |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |
|                        |          |                |                 |                    |                ch_size |         |                |
|------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------|
| _objective_d6e35_00000 | RUNNING  | 172.2

 20%|██        | 59/290 [00:25<01:17,  2.97it/s]


== Status ==
Current time: 2022-11-29 17:26:13 (running for 00:00:34.94)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (2 PENDING, 1 RUNNING)
+------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------+
| Trial name             | status   | loc            |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |
|                        |          |                |                 |                    |                ch_size |         |                |
|------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------|
| _objective_d6e35_00000 | RUNNING  | 172.2

 25%|██▍       | 72/290 [00:30<01:25,  2.54it/s]


== Status ==
Current time: 2022-11-29 17:26:18 (running for 00:00:39.95)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (2 PENDING, 1 RUNNING)
+------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------+
| Trial name             | status   | loc            |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |
|                        |          |                |                 |                    |                ch_size |         |                |
|------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------|
| _objective_d6e35_00000 | RUNNING  | 172.2

 29%|██▉       | 85/290 [00:35<01:20,  2.54it/s]


== Status ==
Current time: 2022-11-29 17:26:23 (running for 00:00:44.95)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (2 PENDING, 1 RUNNING)
+------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------+
| Trial name             | status   | loc            |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |
|                        |          |                |                 |                    |                ch_size |         |                |
|------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------|
| _objective_d6e35_00000 | RUNNING  | 172.2

 33%|███▎      | 97/290 [00:40<01:14,  2.59it/s]


== Status ==
Current time: 2022-11-29 17:26:28 (running for 00:00:49.96)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (2 PENDING, 1 RUNNING)
+------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------+
| Trial name             | status   | loc            |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |
|                        |          |                |                 |                    |                ch_size |         |                |
|------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------|
| _objective_d6e35_00000 | RUNNING  | 172.2

 38%|███▊      | 110/290 [00:45<01:12,  2.49it/s]


== Status ==
Current time: 2022-11-29 17:26:33 (running for 00:00:54.96)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (2 PENDING, 1 RUNNING)
+------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------+
| Trial name             | status   | loc            |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |
|                        |          |                |                 |                    |                ch_size |         |                |
|------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------|
| _objective_d6e35_00000 | RUNNING  | 172.2

 42%|████▏     | 123/290 [00:50<01:07,  2.47it/s]


== Status ==
Current time: 2022-11-29 17:26:38 (running for 00:00:59.97)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (2 PENDING, 1 RUNNING)
+------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------+
| Trial name             | status   | loc            |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |
|                        |          |                |                 |                    |                ch_size |         |                |
|------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------|
| _objective_d6e35_00000 | RUNNING  | 172.2

 47%|████▋     | 135/290 [00:55<01:02,  2.48it/s]


== Status ==
Current time: 2022-11-29 17:26:43 (running for 00:01:04.97)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (2 PENDING, 1 RUNNING)
+------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------+
| Trial name             | status   | loc            |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |
|                        |          |                |                 |                    |                ch_size |         |                |
|------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------|
| _objective_d6e35_00000 | RUNNING  | 172.2

 51%|█████     | 147/290 [01:00<00:57,  2.47it/s]


== Status ==
Current time: 2022-11-29 17:26:48 (running for 00:01:09.98)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (2 PENDING, 1 RUNNING)
+------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------+
| Trial name             | status   | loc            |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |
|                        |          |                |                 |                    |                ch_size |         |                |
|------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------|
| _objective_d6e35_00000 | RUNNING  | 172.2

 55%|█████▍    | 159/290 [01:05<00:53,  2.46it/s]


== Status ==
Current time: 2022-11-29 17:26:53 (running for 00:01:14.98)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (2 PENDING, 1 RUNNING)
+------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------+
| Trial name             | status   | loc            |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |
|                        |          |                |                 |                    |                ch_size |         |                |
|------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------|
| _objective_d6e35_00000 | RUNNING  | 172.2

 59%|█████▉    | 171/290 [01:10<00:49,  2.40it/s]


== Status ==
Current time: 2022-11-29 17:26:58 (running for 00:01:19.99)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (2 PENDING, 1 RUNNING)
+------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------+
| Trial name             | status   | loc            |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |
|                        |          |                |                 |                    |                ch_size |         |                |
|------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------|
| _objective_d6e35_00000 | RUNNING  | 172.2

 63%|██████▎   | 184/290 [01:15<00:47,  2.22it/s]


== Status ==
Current time: 2022-11-29 17:27:03 (running for 00:01:24.99)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (2 PENDING, 1 RUNNING)
+------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------+
| Trial name             | status   | loc            |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |
|                        |          |                |                 |                    |                ch_size |         |                |
|------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------|
| _objective_d6e35_00000 | RUNNING  | 172.2

 67%|██████▋   | 195/290 [01:20<00:41,  2.27it/s]


== Status ==
Current time: 2022-11-29 17:27:08 (running for 00:01:30.00)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (2 PENDING, 1 RUNNING)
+------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------+
| Trial name             | status   | loc            |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |
|                        |          |                |                 |                    |                ch_size |         |                |
|------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------|
| _objective_d6e35_00000 | RUNNING  | 172.2

 71%|███████▏  | 207/290 [01:25<00:34,  2.42it/s]


== Status ==
Current time: 2022-11-29 17:27:13 (running for 00:01:35.00)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (2 PENDING, 1 RUNNING)
+------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------+
| Trial name             | status   | loc            |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |
|                        |          |                |                 |                    |                ch_size |         |                |
|------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------|
| _objective_d6e35_00000 | RUNNING  | 172.2

 75%|███████▌  | 218/290 [01:30<00:30,  2.34it/s]


== Status ==
Current time: 2022-11-29 17:27:18 (running for 00:01:40.01)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (2 PENDING, 1 RUNNING)
+------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------+
| Trial name             | status   | loc            |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |
|                        |          |                |                 |                    |                ch_size |         |                |
|------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------|
| _objective_d6e35_00000 | RUNNING  | 172.2

 79%|███████▉  | 230/290 [01:35<00:25,  2.35it/s]


== Status ==
Current time: 2022-11-29 17:27:23 (running for 00:01:45.01)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (2 PENDING, 1 RUNNING)
+------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------+
| Trial name             | status   | loc            |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |
|                        |          |                |                 |                    |                ch_size |         |                |
|------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------|
| _objective_d6e35_00000 | RUNNING  | 172.2

 83%|████████▎ | 242/290 [01:40<00:20,  2.38it/s]


== Status ==
Current time: 2022-11-29 17:27:28 (running for 00:01:50.02)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (2 PENDING, 1 RUNNING)
+------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------+
| Trial name             | status   | loc            |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |
|                        |          |                |                 |                    |                ch_size |         |                |
|------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------|
| _objective_d6e35_00000 | RUNNING  | 172.2

 87%|████████▋ | 253/290 [01:45<00:16,  2.23it/s]


== Status ==
Current time: 2022-11-29 17:27:33 (running for 00:01:55.02)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (2 PENDING, 1 RUNNING)
+------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------+
| Trial name             | status   | loc            |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |
|                        |          |                |                 |                    |                ch_size |         |                |
|------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------|
| _objective_d6e35_00000 | RUNNING  | 172.2

 91%|█████████ | 264/290 [01:50<00:11,  2.18it/s]


== Status ==
Current time: 2022-11-29 17:27:38 (running for 00:02:00.03)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (2 PENDING, 1 RUNNING)
+------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------+
| Trial name             | status   | loc            |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |
|                        |          |                |                 |                    |                ch_size |         |                |
|------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------|
| _objective_d6e35_00000 | RUNNING  | 172.2

 95%|█████████▍| 275/290 [01:55<00:06,  2.19it/s]


== Status ==
Current time: 2022-11-29 17:27:43 (running for 00:02:05.03)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (2 PENDING, 1 RUNNING)
+------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------+
| Trial name             | status   | loc            |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |
|                        |          |                |                 |                    |                ch_size |         |                |
|------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------|
| _objective_d6e35_00000 | RUNNING  | 172.2

 99%|█████████▊| 286/290 [02:00<00:01,  2.24it/s]


== Status ==
Current time: 2022-11-29 17:27:48 (running for 00:02:10.04)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (2 PENDING, 1 RUNNING)
+------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------+
| Trial name             | status   | loc            |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |
|                        |          |                |                 |                    |                ch_size |         |                |
|------------------------+----------+----------------+-----------------+--------------------+------------------------+---------+----------------|
| _objective_d6e35_00000 | RUNNING  | 172.2

100%|██████████| 290/290 [02:02<00:00,  2.37it/s]


(_objective pid=659) {'train_runtime': 122.152, 'train_samples_per_second': 150.141, 'train_steps_per_second': 2.374, 'train_loss': 0.5605058341190733, 'epoch': 5.0}


 86%|████████▋ | 44/51 [00:00<00:00, 52.40it/s]


100%|██████████| 51/51 [00:00<00:00, 51.80it/s]


== Status ==
Current time: 2022-11-29 17:27:56 (running for 00:02:18.10)
Memory usage on this node: 3.7/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 PENDING, 1 RUNNING, 1 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---

(_objective pid=1063) Some weights of the model checkpoint at distilbert-base-uncased were not used when initializing DistilBertForSequenceClassification: ['vocab_transform.bias', 'vocab_layer_norm.weight', 'vocab_projector.weight', 'vocab_projector.bias', 'vocab_transform.weight', 'vocab_layer_norm.bias']
(_objective pid=1063) - This IS expected if you are initializing DistilBertForSequenceClassification from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
(_objective pid=1063) - This IS NOT expected if you are initializing DistilBertForSequenceClassification from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
(_objective pid=1063) Some weights of DistilBertForSequenceClassification were not initialized from the model checkpoint at distilbert-base-unc

== Status ==
Current time: 2022-11-29 17:28:01 (running for 00:02:23.11)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 PENDING, 1 RUNNING, 1 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---

  7%|▋         | 45/690 [00:06<01:19,  8.07it/s]


== Status ==
Current time: 2022-11-29 17:28:06 (running for 00:02:28.11)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 PENDING, 1 RUNNING, 1 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---

 12%|█▏        | 85/690 [00:11<01:10,  8.58it/s]


== Status ==
Current time: 2022-11-29 17:28:11 (running for 00:02:33.12)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 PENDING, 1 RUNNING, 1 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---

 18%|█▊        | 126/690 [00:16<01:10,  8.00it/s]


== Status ==
Current time: 2022-11-29 17:28:16 (running for 00:02:38.12)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 PENDING, 1 RUNNING, 1 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---

 24%|██▍       | 164/690 [00:21<01:06,  7.93it/s]


== Status ==
Current time: 2022-11-29 17:28:21 (running for 00:02:43.13)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 PENDING, 1 RUNNING, 1 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---

 30%|██▉       | 205/690 [00:26<01:05,  7.46it/s]


== Status ==
Current time: 2022-11-29 17:28:26 (running for 00:02:48.13)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 PENDING, 1 RUNNING, 1 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---

 35%|███▌      | 244/690 [00:31<00:55,  8.02it/s]


== Status ==
Current time: 2022-11-29 17:28:31 (running for 00:02:53.14)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 PENDING, 1 RUNNING, 1 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---

 41%|████▏     | 285/690 [00:36<00:50,  8.09it/s]


== Status ==
Current time: 2022-11-29 17:28:36 (running for 00:02:58.14)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 PENDING, 1 RUNNING, 1 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---

 47%|████▋     | 324/690 [00:41<00:46,  7.93it/s]


== Status ==
Current time: 2022-11-29 17:28:41 (running for 00:03:03.15)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 PENDING, 1 RUNNING, 1 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---

 53%|█████▎    | 365/690 [00:46<00:39,  8.14it/s]


== Status ==
Current time: 2022-11-29 17:28:46 (running for 00:03:08.15)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 PENDING, 1 RUNNING, 1 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---

 59%|█████▊    | 405/690 [00:51<00:34,  8.36it/s]


== Status ==
Current time: 2022-11-29 17:28:51 (running for 00:03:13.16)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 PENDING, 1 RUNNING, 1 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---

 64%|██████▍   | 445/690 [00:56<00:31,  7.82it/s]


== Status ==
Current time: 2022-11-29 17:28:56 (running for 00:03:18.16)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 PENDING, 1 RUNNING, 1 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---

 70%|███████   | 485/690 [01:01<00:26,  7.80it/s]


== Status ==
Current time: 2022-11-29 17:29:01 (running for 00:03:23.17)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 PENDING, 1 RUNNING, 1 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---

 72%|███████▏  | 499/690 [01:02<00:24,  7.95it/s]


(_objective pid=1063) {'loss': 0.6098, 'learning_rate': 5.648612608501682e-07, 'epoch': 2.17}


  0%|          | 0/51 [00:00<?, ?it/s]
(_objective pid=1063) 
 14%|█▎        | 7/51 [00:00<00:00, 63.84it/s]
(_objective pid=1063) 
 27%|██▋       | 14/51 [00:00<00:00, 57.18it/s]
(_objective pid=1063) 
 39%|███▉      | 20/51 [00:00<00:00, 53.81it/s]
(_objective pid=1063) 
 51%|█████     | 26/51 [00:00<00:00, 54.55it/s]
(_objective pid=1063) 
 63%|██████▎   | 32/51 [00:00<00:00, 54.17it/s]
(_objective pid=1063) 
 75%|███████▍  | 38/51 [00:00<00:00, 55.44it/s]
(_objective pid=1063) 
 86%|████████▋ | 44/51 [00:00<00:00, 55.14it/s]


(_objective pid=1063) {'eval_loss': 0.5820257663726807, 'eval_accuracy': 0.6838235294117647, 'eval_f1': 0.8122270742358079, 'eval_runtime': 0.9524, 'eval_samples_per_second': 428.397, 'eval_steps_per_second': 53.55, 'epoch': 2.17}


(_objective pid=1063) 
                                                 
100%|██████████| 51/51 [00:00<00:00, 54.61it/s]
                                               
 76%|███████▌  | 526/690 [01:08<00:19,  8.41it/s]


== Status ==
Current time: 2022-11-29 17:29:08 (running for 00:03:30.89)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 PENDING, 1 RUNNING, 1 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---

 82%|████████▏ | 567/690 [01:13<00:14,  8.36it/s]


== Status ==
Current time: 2022-11-29 17:29:13 (running for 00:03:35.89)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 PENDING, 1 RUNNING, 1 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---

 88%|████████▊ | 609/690 [01:18<00:10,  7.67it/s]


== Status ==
Current time: 2022-11-29 17:29:18 (running for 00:03:40.90)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 PENDING, 1 RUNNING, 1 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---

 94%|█████████▍| 649/690 [01:23<00:05,  8.20it/s]


== Status ==
Current time: 2022-11-29 17:29:23 (running for 00:03:45.90)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 PENDING, 1 RUNNING, 1 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---

100%|██████████| 690/690 [01:28<00:00,  7.76it/s]


(_objective pid=1063) {'train_runtime': 88.9293, 'train_samples_per_second': 123.739, 'train_steps_per_second': 7.759, 'train_loss': 0.6039019045622452, 'epoch': 3.0}
== Status ==
Current time: 2022-11-29 17:29:29 (running for 00:03:51.11)
Memory usage on this node: 3.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 RUNNING, 2 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |

(_objective pid=1846) Some weights of the model checkpoint at distilbert-base-uncased were not used when initializing DistilBertForSequenceClassification: ['vocab_projector.bias', 'vocab_layer_norm.weight', 'vocab_transform.weight', 'vocab_projector.weight', 'vocab_transform.bias', 'vocab_layer_norm.bias']
(_objective pid=1846) - This IS expected if you are initializing DistilBertForSequenceClassification from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
(_objective pid=1846) - This IS NOT expected if you are initializing DistilBertForSequenceClassification from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
(_objective pid=1846) Some weights of DistilBertForSequenceClassification were not initialized from the model checkpoint at distilbert-base-unc

== Status ==
Current time: 2022-11-29 17:29:38 (running for 00:04:00.76)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 RUNNING, 2 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----

  4%|▍         | 61/1377 [00:05<01:35, 13.81it/s]


== Status ==
Current time: 2022-11-29 17:29:43 (running for 00:04:05.77)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 RUNNING, 2 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----

  9%|▉         | 129/1377 [00:10<01:33, 13.41it/s]


== Status ==
Current time: 2022-11-29 17:29:48 (running for 00:04:10.77)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 RUNNING, 2 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----

 14%|█▍        | 193/1377 [00:15<01:28, 13.37it/s]


== Status ==
Current time: 2022-11-29 17:29:53 (running for 00:04:15.78)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 RUNNING, 2 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----

 19%|█▉        | 261/1377 [00:20<01:22, 13.52it/s]


== Status ==
Current time: 2022-11-29 17:29:58 (running for 00:04:20.78)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 RUNNING, 2 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----

 24%|██▎       | 327/1377 [00:25<01:17, 13.50it/s]


== Status ==
Current time: 2022-11-29 17:30:03 (running for 00:04:25.79)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 RUNNING, 2 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----

 28%|██▊       | 391/1377 [00:30<01:15, 13.02it/s]


== Status ==
Current time: 2022-11-29 17:30:08 (running for 00:04:30.79)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 RUNNING, 2 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----

 33%|███▎      | 459/1377 [00:35<01:07, 13.60it/s]


== Status ==
Current time: 2022-11-29 17:30:13 (running for 00:04:35.80)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 RUNNING, 2 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----

  0%|          | 0/51 [00:00<?, ?it/s]
(_objective pid=1846) 
 14%|█▎        | 7/51 [00:00<00:00, 61.77it/s]


(_objective pid=1846) {'loss': 0.5262, 'learning_rate': 1.0146019124505001e-05, 'epoch': 1.09}


(_objective pid=1846) 
 27%|██▋       | 14/51 [00:00<00:00, 53.71it/s]
(_objective pid=1846) 
 39%|███▉      | 20/51 [00:00<00:00, 50.54it/s]
(_objective pid=1846) 
 51%|█████     | 26/51 [00:00<00:00, 51.17it/s]
(_objective pid=1846) 
 63%|██████▎   | 32/51 [00:00<00:00, 50.64it/s]
(_objective pid=1846) 
 75%|███████▍  | 38/51 [00:00<00:00, 52.44it/s]
(_objective pid=1846) 
 86%|████████▋ | 44/51 [00:00<00:00, 52.37it/s]
(_objective pid=1846) 
                                                  
100%|██████████| 51/51 [00:00<00:00, 50.85it/s]
                                               


(_objective pid=1846) {'eval_loss': 0.38265106081962585, 'eval_accuracy': 0.8333333333333334, 'eval_f1': 0.8785714285714286, 'eval_runtime': 1.0131, 'eval_samples_per_second': 402.73, 'eval_steps_per_second': 50.341, 'epoch': 1.09}


 39%|███▉      | 543/1377 [00:44<01:03, 13.22it/s]


== Status ==
Current time: 2022-11-29 17:30:22 (running for 00:04:44.90)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 RUNNING, 2 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----

 44%|████▍     | 611/1377 [00:49<00:58, 13.00it/s]


== Status ==
Current time: 2022-11-29 17:30:27 (running for 00:04:49.91)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 RUNNING, 2 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----

 49%|████▉     | 675/1377 [00:54<00:52, 13.33it/s]


== Status ==
Current time: 2022-11-29 17:30:32 (running for 00:04:54.91)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 RUNNING, 2 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----

 54%|█████▍    | 743/1377 [00:59<00:47, 13.38it/s]


== Status ==
Current time: 2022-11-29 17:30:37 (running for 00:04:59.92)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 RUNNING, 2 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----

 59%|█████▉    | 809/1377 [01:04<00:42, 13.29it/s]


== Status ==
Current time: 2022-11-29 17:30:43 (running for 00:05:04.92)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 RUNNING, 2 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----

 64%|██████▍   | 879/1377 [01:09<00:36, 13.59it/s]


== Status ==
Current time: 2022-11-29 17:30:48 (running for 00:05:09.93)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 RUNNING, 2 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----

 69%|██████▊   | 945/1377 [01:14<00:32, 13.25it/s]


== Status ==
Current time: 2022-11-29 17:30:53 (running for 00:05:14.93)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 RUNNING, 2 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----

  0%|          | 0/51 [00:00<?, ?it/s]
(_objective pid=1846) 
 14%|█▎        | 7/51 [00:00<00:00, 62.00it/s]


(_objective pid=1846) {'loss': 0.3433, 'learning_rate': 4.361515632768968e-06, 'epoch': 2.18}


(_objective pid=1846) 
 27%|██▋       | 14/51 [00:00<00:00, 56.53it/s]
(_objective pid=1846) 
 39%|███▉      | 20/51 [00:00<00:00, 53.25it/s]
(_objective pid=1846) 
 51%|█████     | 26/51 [00:00<00:00, 54.32it/s]
(_objective pid=1846) 
 63%|██████▎   | 32/51 [00:00<00:00, 54.46it/s]
(_objective pid=1846) 
 75%|███████▍  | 38/51 [00:00<00:00, 55.60it/s]
(_objective pid=1846) 
 86%|████████▋ | 44/51 [00:00<00:00, 55.11it/s]
(_objective pid=1846) 
                                                   
100%|██████████| 51/51 [00:00<00:00, 54.74it/s]
                                               


(_objective pid=1846) {'eval_loss': 0.4225529730319977, 'eval_accuracy': 0.8455882352941176, 'eval_f1': 0.891566265060241, 'eval_runtime': 0.9505, 'eval_samples_per_second': 429.253, 'eval_steps_per_second': 53.657, 'epoch': 2.18}
== Status ==
Current time: 2022-11-29 17:30:58 (running for 00:05:19.95)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 RUNNING, 2 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |          

 76%|███████▌  | 1047/1377 [01:24<00:24, 13.35it/s]


== Status ==
Current time: 2022-11-29 17:31:03 (running for 00:05:24.95)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 RUNNING, 2 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----

 81%|████████  | 1115/1377 [01:29<00:19, 13.26it/s]


== Status ==
Current time: 2022-11-29 17:31:08 (running for 00:05:29.96)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 RUNNING, 2 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----

 86%|████████▌ | 1179/1377 [01:34<00:14, 13.29it/s]


== Status ==
Current time: 2022-11-29 17:31:13 (running for 00:05:34.97)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 RUNNING, 2 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----

 91%|█████████ | 1249/1377 [01:39<00:09, 13.05it/s]


== Status ==
Current time: 2022-11-29 17:31:18 (running for 00:05:39.97)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 RUNNING, 2 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----

 95%|█████████▌| 1315/1377 [01:44<00:04, 13.52it/s]


== Status ==
Current time: 2022-11-29 17:31:23 (running for 00:05:44.97)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 1.0/8 CPUs, 1.0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (1 RUNNING, 2 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----

100%|██████████| 1377/1377 [01:49<00:00, 12.58it/s]
2022-11-29 17:31:27,817	INFO tune.py:778 -- Total run time: 350.36 seconds (349.63 seconds for the tuning loop).


== Status ==
Current time: 2022-11-29 17:31:27 (running for 00:05:49.63)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 0/8 CPUs, 0/1 GPUs, 0.0/30.18 GiB heap, 0.0/15.09 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-29_17-25-37
Number of trials: 3/3 (3 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+--

In [7]:
print(best_run)

BestRun(run_id='d6e35_00002', objective=1.7371545003543587, hyperparameters={'learning_rate': 1.5930522616241033e-05, 'num_train_epochs': 3, 'seed': 1.8027952775362954, 'per_device_train_batch_size': 8, 'weight_decay': 0.3})


* Result of running Colab Pro +, GPU/High RAM:


```

2022-11-28 02:28:06,252	INFO tune.py:778 -- Total run time: 347.71 seconds (347.41 seconds for the tuning loop).
== Status ==
Current time: 2022-11-28 02:28:06 (running for 00:05:47.41)
Memory usage on this node: 4.4/51.0 GiB 
Using FIFO scheduling algorithm.
Resources requested: 0/8 CPUs, 0/1 GPUs, 0.0/30.17 GiB heap, 0.0/15.08 GiB objects (0.0/1.0 accelerator_type:T4)
Result logdir: /root/ray_results/_objective_2022-11-28_02-22-18
Number of trials: 3/3 (3 TERMINATED)
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+
| Trial name             | status     | loc             |   learning_rate |   num_train_epochs |   per_device_train_bat |    seed |   weight_decay |   objective |
|                        |            |                 |                 |                    |                ch_size |         |                |             |
|------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------|
| _objective_7b66a_00000 | TERMINATED | 172.28.0.2:655  |     5.61152e-06 |                  5 |                     64 | 8.15396 |            0.1 |     1.5247  |
| _objective_7b66a_00001 | TERMINATED | 172.28.0.2:1060 |     2.05134e-06 |                  3 |                     16 | 4.89902 |            0.1 |     1.49605 |
| _objective_7b66a_00002 | TERMINATED | 172.28.0.2:1841 |     1.59305e-05 |                  3 |                      8 | 1.8028  |            0.2 |     1.72919 |
+------------------------+------------+-----------------+-----------------+--------------------+------------------------+---------+----------------+-------------+


(_objective pid=1841) {'train_runtime': 107.4607, 'train_samples_per_second': 102.4, 'train_steps_per_second': 12.814, 'train_loss': 0.3774486755060821, 'epoch': 3.0}
BestRun(run_id='7b66a_00002', objective=1.7291939932062017, hyperparameters={'learning_rate': 1.5930522616241033e-05, 'num_train_epochs': 3, 'seed': 1.8027952775362954, 'per_device_train_batch_size': 8, 'weight_decay': 0.2})
```



##HPO Using Optuna

In [8]:
from typing import Dict

def hp_space_optuna(trial) -> Dict[str, float]:
    return {
        "learning_rate" : trial.suggest_float("learning_rate", 1e-6, 1e-4, log=True),
        "num_train_epochs" : trial.suggest_int("num_train_epochs", 1, 5),
        "seed" : trial.suggest_int("seed", 1, 40),
        "per_device_train_batch_size" : trial.suggest_categorical("per_device_train_batch_size", [4, 8, 16, 32, 64]),
        "weight_decay" : trial.suggest_loguniform('weight_decay', 0.001, 0.1)
    }


In [9]:
!pip install optuna

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 348 kB 22.3 MB/s 
     |████████████████████████████████| 209 kB 86.9 MB/s 
     |████████████████████████████████| 81 kB 10.8 MB/s 
     |████████████████████████████████| 78 kB 7.9 MB/s 
     |████████████████████████████████| 112 kB 73.7 MB/s 
     |████████████████████████████████| 147 kB 84.6 MB/s 
     |████████████████████████████████| 50 kB 7.4 MB/s 
  Created wheel for pyperclip: filename=pyperclip-1.8.2-py3-none-any.whl size=11136 sha256=9f3ee6e81e695bc415c66bae0cb4d645ff089c0a61f7274c194a37e38bcbc88f
  Stored in directory: /root/.cache/pip/wheels/9f/18/84/8f69f8b08169c7bae2dde6bd7daf0c19fca8c8e500ee620a28
Successfully built pyperclip


In [10]:
import optuna

In [11]:
%%time
if RUN == RUN_OPTUNA or RUN == RUN_ALL:
  best_run = trainer.hyperparameter_search(
      hp_space=hp_space_optuna,
      direction="maximize",
      backend="optuna",
      n_trials=3
  )
  print("------------OPTUNA------------")
  print(best_run)

[I 2022-11-29 17:31:39,523] A new study created in memory with name: no-name-d173dd73-612d-45b8-a0a6-9c5d11025337
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:9: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  if __name__ == '__main__':
Trial: {'learning_rate': 2.9378274943857116e-05, 'num_train_epochs': 5, 'seed': 27, 'per_device_train_batch_size': 64, 'weight_decay': 0.02237594008580847}
loading configuration file config.json from cache at /root/.cache/huggingface/hub/models--distilbert-base-uncased/snapshots/1c4513b2eedbda136f57676a34eea67aba266e5c/config.json
Model config DistilBertConfig {
  "_name_or_path": "distilbert-base-uncased",
  "activation": "gelu",
  "architectures": [
    "DistilBertForMaskedLM"
  ],
  "attention_dropout": 0.1,
  "dim": 768,
  "dropout": 0.1,
  "hidden_dim": 3072,
  "i



Training completed. Do not forget to share your model on huggingface.co/models =)


The following columns in the evaluation set don't have a corresponding argument in `DistilBertForSequenceClassification.forward` and have been ignored: sentence2, sentence1, idx. If sentence2, sentence1, idx are not expected by `DistilBertForSequenceClassification.forward`,  you can safely ignore this message.
***** Running Evaluation *****
  Num examples = 408
  Batch size = 8


[I 2022-11-29 17:33:50,810] Trial 0 finished with value: 1.7224171394446104 and parameters: {'learning_rate': 2.9378274943857116e-05, 'num_train_epochs': 5, 'seed': 27, 'per_device_train_batch_size': 64, 'weight_decay': 0.02237594008580847}. Best is trial 0 with value: 1.7224171394446104.
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:9: FutureWarning: suggest_loguniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use :func:`~optuna.trial.Trial.suggest_float` instead.
  if __name__ == '__main__':
Trial: {'learning_rate': 6.155893160017104e-06, 'num_train_epochs': 2, 'seed': 12, 'per_device_train_batch_size': 8, 'weight_decay': 0.039499246466172394}
loading configuration file config.json from cache at /root/.cache/huggingface/hub/models--distilbert-base-uncased/snapshots/1c4513b2eedbda136f57676a34eea67aba266e5c/config.json
Model config DistilBertConfig {
  "_name_or_path": "distilbert-base

The following columns in the evaluation set don't have a corresponding argument in `DistilBertForSequenceClassification.forward` and have been ignored: sentence2, sentence1, idx. If sentence2, sentence1, idx are not expected by `DistilBertForSequenceClassification.forward`,  you can safely ignore this message.
***** Running Evaluation *****
  Num examples = 408
  Batch size = 8
Saving model checkpoint to test/run-1/checkpoint-500
Configuration saved in test/run-1/checkpoint-500/config.json
Model weights saved in test/run-1/checkpoint-500/pytorch_model.bin
tokenizer config file saved in test/run-1/checkpoint-500/tokenizer_config.json
Special tokens file saved in test/run-1/checkpoint-500/special_tokens_map.json


Training completed. Do not forget to share your model on huggingface.co/models =)


[I 2022-11-29 17:35:03,788] Trial 1 finished with value: 1.541485949218397 and parameters: {'learning_rate': 6.155893160017104e-06, 'num_train_epochs': 2, 'seed': 12, 'per_device_train_batch_siz

The following columns in the evaluation set don't have a corresponding argument in `DistilBertForSequenceClassification.forward` and have been ignored: sentence2, sentence1, idx. If sentence2, sentence1, idx are not expected by `DistilBertForSequenceClassification.forward`,  you can safely ignore this message.
***** Running Evaluation *****
  Num examples = 408
  Batch size = 8
Saving model checkpoint to test/run-2/checkpoint-500
Configuration saved in test/run-2/checkpoint-500/config.json
Model weights saved in test/run-2/checkpoint-500/pytorch_model.bin
tokenizer config file saved in test/run-2/checkpoint-500/tokenizer_config.json
Special tokens file saved in test/run-2/checkpoint-500/special_tokens_map.json


Training completed. Do not forget to share your model on huggingface.co/models =)


[I 2022-11-29 17:36:16,840] Trial 2 finished with value: 1.528846153846154 and parameters: {'learning_rate': 7.2967030975123275e-06, 'num_train_epochs': 2, 'seed': 2, 'per_device_train_batch_siz

------------
BestRun(run_id='0', objective=1.7224171394446104, hyperparameters={'learning_rate': 2.9378274943857116e-05, 'num_train_epochs': 5, 'seed': 27, 'per_device_train_batch_size': 64, 'weight_decay': 0.02237594008580847})
CPU times: user 4min 55s, sys: 4.23 s, total: 4min 59s
Wall time: 4min 37s


* optuna finished result:
* [I 2022-11-28 08:38:32,613] Trial 2 finished with value: 1.7188552188552189 and parameters: {'learning_rate': 3.9534652778264654e-05, 'num_train_epochs': 3, 'seed': 15, 'per_device_train_batch_size': 32, 'weight_decay': 0.0023147682141046017}. 
* Best is trial 2 with value: 1.7188552188552189.
* CPU times: user 7min 53s, sys: 10.2 s, total: 8min 3s
Wall time: 8min 4s
* BestRun(run_id='2', objective=1.7188552188552189, hyperparameters={'learning_rate': 3.9534652778264654e-05, 'num_train_epochs': 3, 'seed': 15, 'per_device_train_batch_size': 32, 'weight_decay': 0.0023147682141046017}) 
* 10/28 BestRun(run_id='0', objective=1.6687864334923157, hyperparameters={'learning_rate': 6.756272182144086e-06, 'num_train_epochs': 3, 'seed': 21, 'per_device_train_batch_size': 8, 'weight_decay': 0.0017858013659287576})
* 10/29 BestRun(run_id='0', objective=1.7224171394446104, hyperparameters={'learning_rate': 2.9378274943857116e-05, 'num_train_epochs': 5, 'seed': 27, 'per_device_train_batch_size': 64, 'weight_decay': 0.02237594008580847})
* 10/29 CPU times: user 4min 55s, sys: 4.23 s, total: 4min 59s
* Wall time: 4min 37s



In [12]:
print(best_run)

BestRun(run_id='0', objective=1.7224171394446104, hyperparameters={'learning_rate': 2.9378274943857116e-05, 'num_train_epochs': 5, 'seed': 27, 'per_device_train_batch_size': 64, 'weight_decay': 0.02237594008580847})
